inc/Config/DataSource/HttpDataSource.php

```diff
@@ -98,11 +98,51 @@ protected static function get_service_config_schema(): array {
 		return ConfigSchemas::get_http_data_source_service_config_schema();
 	}
 
+	private static function get_auth_to_request_headers( array $service_config ): array {
+		if ( ! isset( $service_config['auth'] ) ) {
+			return [];
+		}
+
+		$auth_headers = [];
+		$auth = $service_config['auth'];
+
+		if ( 'bearer' === $auth['type'] ) {
+			$auth_headers['Authorization'] = 'Bearer ' . $auth['value'];
+		} elseif ( 'basic' === $auth['type'] ) {
+			$auth_headers['Authorization'] = 'Basic ' . base64_encode( $auth['value'] );
+		} elseif ( 'api-key' === $auth['type'] && 'header' === $auth['add_to'] ) {
+			$auth_headers[ $auth['key'] ] = $auth['value'];
+		}
+
+		return $auth_headers;
+	}
+
+	private static function add_auth_to_endpoint( string|null $endpoint, array $service_config ): string|null {
+		if ( ! isset( $service_config['auth'] ) 
+			|| 'api-key' !== $service_config['auth']['type'] 
+			|| 'queryparams' !== $service_config['auth']['add_to']
+			|| empty( $endpoint )
+		) {
+			return $endpoint;
+		}
+
+		$query_string = $service_config['auth']['key'] . '=' . $service_config['auth']['value'];
+		
+		return $endpoint . ( strpos( $endpoint, '?' ) === false ? '?' : '&' ) . $query_string;
```

Comment:
It might be time to use some built in methods from WordPress or PHP to do URL construction rather than doing it by hand.